bundle/deployplan/plan.go

```diff
@@ -0,0 +1,66 @@
+package deployplan
+
+import (
+	"cmp"
+	"slices"
+	"strings"
+)
+
+type Plan struct {
+	// Current version is zero which is omitted and has no backward compatibility guarantees
+	PlanVersion int `json:"plan_version,omitempty"`
+	// TODO:
+	// - CliVersion  string               `json:"cli_version"`
+	// - Copy Serial / Lineage from the state file
+	Plan map[string]PlanEntry `json:"plan,omitzero"`
+}
+
+type PlanEntry struct {
+	ID        string           `json:"id,omitempty"`
+	DependsOn []DependsOnEntry `json:"depends_on,omitempty"`
+	Action    string           `json:"action"`
+	Fields    []Field          `json:"fields,omitempty"`
+}
+
+type DependsOnEntry struct {
+	Node  string `json:"node"`
+	Label string `json:"label,omitempty"`
+}
+
+type Field struct {
+	Path   string `json:"path"`
+	State  any    `json:"state,omitempty"`
+	Config any    `json:"config"`
+	Remote any    `json:"remote,omitempty"`
+	Action string `json:"action"`
+}
+
+func (p Plan) GetActions() []Action {
+	actions := make([]Action, 0, len(p.Plan))
+	for key, entry := range p.Plan {
+		at := ActionTypeFromString(entry.Action)
+		if at == ActionTypeUnset {
+			continue
+		}
+		if entry.Action == ActionTypeNoop.StringFull() {
```

Comment:
This can now be folded into the previous conditional, `at == ActionTypeNoop`

Reply:
Thanks. I dropped this check completely f08f85b1a29db93d00c9e7f99eae71c911e0f652

Since it's merely different representation of Plan, it should be as lossless as possible.